### PR TITLE
Only use XunitTheoryTestCaseRunner with XunitTheoryTestCase (#245)

### PR DESCRIPTION
**Pull request by cesarbs**
Fix for #245.

cc @pranavkm @NTaylorMullen @rynowak

### CODE REVIEW

File: src/Microsoft.AspNetCore.Testing/xunit/ConditionalAttributeDiscoverer.cs [hunk 1]
```diff
@@ -41,7 +39,7 @@ public ConditionalAttributeDiscoverer(IMessageSink diagnosticMessageSink)
                 .Discover(discoveryOptions, testMethod, attributeInfo)
                 .Select(testCase => new SkipReasonTestCase
                 {
-                    IsTheory = isTheory,
+                    IsTheory = testCase.GetType() == typeof(XunitTheoryTestCase),
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Testing/xunit/ConditionalAttributeDiscoverer.cs]: Might want to do an IsAssignableFrom here just incase other systems subclass.

**Comment by cesarbs** [comment]
Tested this with Kestrel comparing test summaries before and after the move to MSBuild, and with this change:

**Before MSBuild**

Microsoft.AspNetCore.Server.Kestrel.FunctionalTests  Total: 230, Errors: 0, Failed: 0, Skipped: 1, Time: 60.438s
Microsoft.AspNetCore.Server.Kestrel.FunctionalTests  Total: 230, Errors: 0, Failed: 0, Skipped: 1, Time: 30.741s
Microsoft.AspNetCore.Server.KestrelTests  Total: 1357, Errors: 0, Failed: 0, Skipped: 0, Time: 9.266s
Microsoft.AspNetCore.Server.KestrelTests  Total: 1359, Errors: 0, Failed: 0, Skipped: 0, Time: 14.549s

**After MSBuild**

Total tests: 320. Passed: 319. Failed: 0. Skipped: 1.
Total tests: 320. Passed: 319. Failed: 0. Skipped: 1.
Total tests: 1357. Passed: 1357. Failed: 0. Skipped: 0.
Total tests: 1359. Passed: 1359. Failed: 0. Skipped: 0.

**With this PR**

Total tests: 230. Passed: 229. Failed: 0. Skipped: 1.
Total tests: 230. Passed: 229. Failed: 0. Skipped: 1.
Total tests: 1357. Passed: 1357. Failed: 0. Skipped: 0.
Total tests: 1359. Passed: 1359. Failed: 0. Skipped: 0.

**Comment by halter73** [comment]
Is it possible to test the test fix?

File: src/Microsoft.AspNetCore.Testing/xunit/ConditionalAttributeDiscoverer.cs (hunk 1 above)
Comment (halter73) [inline src/Microsoft.AspNetCore.Testing/xunit/ConditionalAttributeDiscoverer.cs]: 👍 Out of curiosity, is that a think that actually happens?

Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Testing/xunit/ConditionalAttributeDiscoverer.cs]: Eh, I haven't seen it in our repos but it's a supported scenario from XUnit. Using `IsAssignableFrom` here would just be for sake of correctness in-case we happen to do it in the future.

Comment (cesarbs) [inline src/Microsoft.AspNetCore.Testing/xunit/ConditionalAttributeDiscoverer.cs]: I don't seem able to use this in .NET Core.

**Comment by cesarbs** [comment]
@halter73 Added test.

File: src/Microsoft.AspNetCore.Testing/xunit/ConditionalAttributeDiscoverer.cs (hunk 1 above)
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Testing/xunit/ConditionalAttributeDiscoverer.cs]: `GetTypeInfo().IsAssignableFrom`

Comment (cesarbs) [inline src/Microsoft.AspNetCore.Testing/xunit/ConditionalAttributeDiscoverer.cs]: Nvm, figured it out.

File: test/Microsoft.AspNetCore.Testing.Tests/ConditionalTheoryTest.cs [hunk 2]
```diff
@@ -15,5 +14,15 @@ public void ConditionalTheorySkip(int arg)
...
         }
+
+        private static int _conditionalTheoryRuns = 0;
+        [ConditionalTheory]
+        [InlineData(0)]
+        [InlineData(1)]
+        public void ConditionalTheoryRunOncePerDataLine(int arg)
+        {
+            _conditionalTheoryRuns++;
+            Assert.True(_conditionalTheoryRuns <= 2, $"Theory should run 2 times, but ran {_conditionalTheoryRuns} times.");
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Testing.Tests/ConditionalTheoryTest.cs]: :+1:

**Pull request merged**